src/Utils/Utilities.cs

```diff
@@ -651,6 +650,48 @@ public static string HashSHA256(byte[] raw)
         return BytesToHex(SHA256.HashData(raw));
     }
 
+    /// <summary>Computes the SHA 256 hash of a file stream and returns it as plaintext.</summary>
+    public static string HashSHA256(FileStream fileStream)
+    {
+        return BytesToHex(SHA256.HashData(fileStream));
+    }
+
+    /// <summary>Checks if a provided string is a valid SHA256 hash in HEX string format, optionally not requiring a '0x' hex prefix.</summary>
+    public static bool IsValidSHA256Hash(string hashHexString, bool requireHexPrefix = true)
+    {
+        if (hashHexString is null)
+        {
+            Logs.Debug($"Invalid SHA256 string, was null");
```

Comment:
this should be `Verbose` if shown at all? Or a return string for if errors are needed by a caller. A Utility method should minimize side-effect behavior